fMain.cs

```diff
@@ -669,19 +700,32 @@ private void TryRandomize(BackgroundWorker worker, DoWorkEventArgs e)
             }
 
             // Additional validation of preconditions
-            if (!ValidateInputFile()) return;
-
-            if (!ValidateROM(Settings.InputROMFilename))
+            if (Settings.GenerateROM)
             {
-                MessageBox.Show("Cannot verify input ROM is Majora's Mask (U).",
-                    "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
-                return;
-            }
+                if (!ValidateInputFile()) return;
 
-            MakeROM(Settings.InputROMFilename, Settings.OutputROMFilename, worker);
+                if (!ValidateROM(Settings.InputROMFilename))
+                {
+                    MessageBox.Show("Cannot verify input ROM is Majora's Mask (U).",
+                        "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
+                    return;
+                }
+            }
+            
+            
 
-            MessageBox.Show("Successfully built output ROM!",
-                "Success", MessageBoxButtons.OK, MessageBoxIcon.None);
+            if (Settings.GenerateROM)
+            {
+                MakeROM(Settings.InputROMFilename, Settings.OutputROMFilename, worker);
+                MessageBox.Show("Successfully built output ROM!",
+                    "Success", MessageBoxButtons.OK, MessageBoxIcon.None);
+            }
+            else

```

Comment:
As i mentioned in another comment, this shouldn't be `else` it should be `if (Settings.GenerateSpoilerLog)` and remove WriteSpoilerLog from the MakeROM function.